### ISSUE TITLE
html snippets not working

### ISSUE THREAD

**Issue by mkotsovoulou** [Codiad/Codiad#1118]
Hello!
I am new to Codiad but I find it extremely useful...
I have a problem making html snippets to work... I have installed code-compelte-plus and autosuggestions do work for CSS and PHP files, as well as code snippets for PHP.
While editing a new html file, I try to type html5 and press TAB, but the snippet (which exists in html.snippets) does not come up...

What am I doing wrong?

**Comment by HLSiira** [comment]
Are you by chance creating the html file while codiad is set to another coding language?

I've haven't used the CodeCompletePlus a lot, but in trying it out, it seems that it's language specific and requires that you are editing an html file, or have your language at the bottom bar of Codiad set to HTML before it will show you html snippets.

By having mine set to html, I was able to have the html5 autocomplete show up.

**Comment by mkotsovoulou** [comment]
Of course, I am creating an html file. I tried with htm but the same...
At random times it might work...
It seems that there is an issue with "understanding" the tab key...
I tried with all browsers :-(
I checked the console for an errors (but nothing)

Anyone facing the same problem?

**Comment by HLSiira** [comment]
I tried looking though the code a little bit and it honestly seems like the plugin itself needs some performance polishing so that might the problem. I'll spend sometime this weekend trying to clean it up a bit and send you a copy but I can't update the plugin repo, and my main focus has been updating my own Codiad Fork, Atheos.you should check it out, I've been doing an insane amount of work on it.

https://github.com/HLSiira/Atheos

**Comment by mkotsovoulou** [comment]
Really??

I will check it out!

I am a college professor and my students struggle with maintaining their files… Editing on the server its been a charm…
The only problem was the not all features where functioning well…


Thanks!

Maira

**Comment by HLSiira** [comment]
I'm doing everything I can to fix as many bugs and features while still maintaining the original compatibiilty but some things to do get lost along the way. Feel free to put in issues in the Atheos fork as you come across them.

**Comment by mkotsovoulou** [comment]
I downloaded Atheos and performed setup as I have for Codiad … but I get many errors … The first and most important one was that it was missing css files…

Best,
Maira

**Comment by HLSiira** [comment]
Man I need to figure out why it keeps doing that.
I'm willing to bet it's still trying to set the theme to "default" in your config.php. If you open that up and change it to "atheos", and then clear cache and cookies, it should load the new updated theme and work from there.

Please let me know if that doesn't work.

**Comment by HLSiira** [comment]
Okay so update for CompletePlus.

I'm still trying to familiarize myself with the plugin itself, but I have an idea of what could be the cause. Codiad has it's own built in AutoComplete component that might be overriding the plugin. Occasionally the plugin won't work for me and sometimes it will, when it doesn't I still get an autocomplete window that follow more like what the built in feature provides:

The default autocomplete seems to only look at the open editor, create a word cloud and make suggestions based on that, while the CompletePlus has it's own library of suggestions and snippets I think for it to pull from.

I'm still looking into the issue though.

**Comment by HLSiira** [comment]
Alright I have a fix in mind, hopefully I'll have a repo for you to use by tomorrow night for the base Codaid installation; however I can definitively say that Atheos will have a working AutoComplete probably sooner than that.

**Comment by mkotsovoulou** [comment]
Wow!!! Thanks!

**Comment by HLSiira** [comment]
While I highly recommend that you update to using Atheos since Codiad is un-maintained and Atheos has at least one very active person (me), I think I managed to make a plugin that should fix the default Codiad auto complete issue.

https://github.com/HLSiira/Codiad-CompletePlus

Atheos will have this by default next push to master btw.

Let me know if you have any issues going further. Always happy to try to help.
